src/webui.c

```diff
@@ -56,6 +56,35 @@ static struct event_base *webui_base;
 static struct evhttp *webui_httpd;
 static struct evhttp_bound_socket *webui_listener;
 
+static void
+send_json_workers(struct evhttp_request *req, void *arg)
+{
+    struct evbuffer *buf = evhttp_request_get_output_buffer(req);
+    struct evkeyvalq *hdrs_in = NULL;
+    struct evkeyvalq *hdrs_out = NULL;
+    char rig_list[MAX_RIG_ID*MAX_RIGS+(MAX_RIGS*3)] = {0};
+    
+    hdrs_in = evhttp_request_get_input_headers(req);
+    const char *cookies = evhttp_find_header(hdrs_in, "Cookie");
+    
+    if (cookies)
+    {
+        char *wa = strstr(cookies, "wa=");
+        if (wa)
+        {
+            char *sc = strchr(wa, ';');
+            if (sc)
+                *sc = 0;
+            wa += 3;
```

Comment:
As there are now two functions that want to extract the address from the header cookie, I'd suggest refactoring that to its own function.

Reply:
moved into const char *fetch_wa_cookie(req); in 861f9da